ember/app/services/fetch.js

```diff
@@ -0,0 +1,53 @@
+import Service, { inject as service } from "@ember/service";
+import { isEmpty } from "@ember/utils";
+import fetch from "fetch";
+
+const CONTENT_TYPE = "application/vnd.api+json";
+
+const cleanObject = (obj) => {
+  return Object.entries(obj).reduce((clean, [key, value]) => {
+    return {
+      ...clean,
+      ...(isEmpty(value) ? {} : { [key]: value }),
+    };
+  }, {});
+};
+
+export default class FetchService extends Service {
+  @service session;
+
+  async fetch(resource, init = {}) {
```

Comment:
I would go with the established naming from `fetch`: `fetch(resource, options)`.

Reply:
Done